esmvalcore/preprocessor/_regrid.py

```diff
@@ -252,15 +252,64 @@ def regrid(cube, target_grid, scheme, lat_offset=True, lon_offset=True):
                 [coord] = coords
                 cube.remove_coord(coord)
 
-    # Perform the horizontal regridding.
-    if _attempt_irregular_regridding(cube, scheme):
-        cube = esmpy_regrid(cube, target_grid, scheme)
-    else:
-        cube = cube.regrid(target_grid, HORIZONTAL_SCHEMES[scheme])
+    # Return unaltered cube if hotizontal grid is the same.
+    if not _check_horiz_grid_closeness(cube, target_grid):
+
+        # Perform the horizontal regridding.
+        if _attempt_irregular_regridding(cube, scheme):
+            cube = esmpy_regrid(cube, target_grid, scheme)
+        else:
+            cube = cube.regrid(target_grid, HORIZONTAL_SCHEMES[scheme])
 
     return cube
 
 
+def _check_horiz_grid_closeness(cube1, cube2):
+    """
+    Check if two cubes have the same horizontal grid definition.
+
+    The result of the function is a boolean answer, if both cubes have the
+    same horizontal grid definition. The function checks both longitude and
+    latitude, based on extent and resolution.
+
+    Parameters
+    ----------
+    cube1 : cube
+        The first of the cubes to be checked.
+    cube2 : cube
+        The second of the cubes to be checked.
+
+    Returns
+    -------
+    bool
+
+    .. note::
+
+        The current implementation checks if the bounds and the
+        grid shapes are the same.
+        Exits on first difference.
+    """
+    # Go through the 2 expected horizontal coordinates longitude and latitude.
+    for coord in ['latitude', 'longitude']:
+        # Compare shapes.
+        if cube1.coord(coord).shape != cube2.coord(coord).shape:
+            # Different shapes detected.
+            return False
+        else:
+            # Iterate through the combined bounds.
```

Comment:
what about the points values, shouldn't those be equal too?

Reply:
Don't need to be. Assume the points are upper left coordinates, so bounds are more precise here.